jujugui/static/gui/src/app/assets/javascripts/react-click-outside.js

```diff
@@ -21,10 +21,23 @@ function enhanceWithClickOutside(WrappedComponent) {
     componentDidMount: function componentDidMount() {
       this.__wrappedComponent = this.refs.wrappedComponent;
       document.addEventListener('click', this.handleClickOutside, true);
+      // XXX ant 9 Feb 2016 Add Esc keybind to also close the current component
+      document.addEventListener('keydown', this.handleKeyDown);
     },
 
     componentWillUnmount: function componentWillUnmount() {
       document.removeEventListener('click', this.handleClickOutside, true);
+      document.removeEventListener('keydown', this.handleKeyDown);
+    },
+
+    handleKeyDown: function handleKeyDown(e) {
```

Comment:
might be worth a comment here that you added this keydown code as an addon to the one we used originally.